src/npctalk.cpp

```diff
@@ -897,7 +909,14 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
         }
         std::vector<skill_id> trainable = p->skills_offered_to( g->u );
         std::vector<matype_id> styles = p->styles_offered_to( g->u );
-        if( trainable.empty() && styles.empty() ) {
+        const std::vector<spell_id> spells = p->magic.spells();
+        std::vector<spell_id> teachable_spells;
+        for( const spell_id &sp : spells ) {
+            if( g->u.magic.can_learn_spell( g->u, sp ) ) {
+                teachable_spells.push_back( sp );
```

Comment:
```suggestion
                teachable_spells.emplace_back( sp );
```
emplace_back is almost always better than push_back